packages/synapse-react-client/src/components/DraggableDialog/DraggableDialog.tsx

```diff
@@ -23,52 +25,97 @@ export default function DraggableDialog({
   title,
   children,
 }: DraggableDialogProps) {
+  const theme = useTheme()
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'))
   const draggableRef = useRef<HTMLDivElement>(null)
+  const [bounds, setBounds] = useState<DraggableBounds>()
   const position = { x: 100, y: 100 }
 
+  useEffect(() => {
+    function updateBounds() {
+      if (draggableRef.current) {
+        const { offsetWidth, offsetHeight } = draggableRef.current
+        const margin = 100
+
+        setBounds({
+          left: -(offsetWidth - margin),
+          top: -(offsetHeight - margin),
+          right: window.innerWidth - margin,
+          bottom: window.innerHeight - margin,
+        })
+      }
+    }
+    updateBounds()
+    window.addEventListener('resize', updateBounds)
+    return () => window.removeEventListener('resize', updateBounds)
+  }, [])
+
   if (!open) {
     return null
   }
 
+  const paperContent = (
+    <Paper
+      ref={draggableRef}
+      elevation={5}
+      sx={{
+        display: 'flex',
+        flexDirection: 'column',
+        height: '500px',
+        width: '600px',
+        ...(isMobile && {
+          position: 'fixed',
+          top: '50%',
+          left: '50%',
+          transform: 'translate(-50%, -50%)',
+          width: '90vw',
+          height: '80vh',
+          maxWidth: '600px',
+          maxHeight: '500px',
+        }),
+      }}
+    >
+      <Stack
+        direction="row"
+        alignItems="center"
+        gap="5px"
+        sx={{
+          padding: '20px',
+        }}
+      >
+        <Typography variant="headline1">{title}</Typography>
+        <Box sx={{ flexGrow: 1 }} />
+        <IconButton onClick={onClose}>
+          <CloseIcon />
+        </IconButton>
+      </Stack>
+      <Divider sx={{ mx: 2 }} />
+      <Box
+        sx={{
+          height: '100%',
+          maxWidth: '100%',
+          padding: '16px',
+          overflow: 'hidden',
+        }}
+      >
+        {children}
+      </Box>
```

Comment:
Seems like this could be a `DialogContent`, which already gracefully handles overflow